src/Withdrawal.java

```diff
@@ -117,7 +117,10 @@ private int displayMenuOfAmounts()
             case 3: // corresponding amount from amounts array
             case 4:
             case 5:
-               userChoice = amounts[ input ]; // save user's choice
+               // userChoice = amounts[ input ]; // save user's choice
+               screen.displayMessage("\nHow much do you want to withdraw? ");
+               int customAmount = keypad.getInput();
+               userChoice = customAmount;
                break;       
             case CANCELED: // the user chose to cancel
                userChoice = CANCELED; // save user's choice
```

Comment:
Re-use an existing variable in a different context is not a best practice. In this case, userChoice is used also to store the amount. This could potentially cause unexpected behaviour. And in fact, with this implementation you cannot withdraw 6 dollars, because it’s the same number as the userChoice CANCELED.